### DISCUSSION TITLE
Ocean Current Turbine Calculation

### DISCUSSION THREAD

**Discussion by YuYuJou** [OpenFAST/openfast#2528] [Q&A] [answered]
I need to simulate an ocean current turbine, I want to ask how to convert Aeroblade to Hydroblade.
Thank you to those who are willing to answer my questions.

**Comment by jjonkman** [comment, marked as answer]
Dear @YuYuJou,

NREL is in the process of adding functionality in OpenFAST to support marine hydrokinetic (MHK) (ocean current) turbines.  This is accomplished, not by modeling blades in HydroDyn, but rather adding hydrodynamic capability to AeroDyn.  You'll see the current version of AeroDyn already supports buoyancy and a cavitation check.  Coming soon in future releases of OpenFAST (in v5) are additional functionality, including hydrodynamic added mass (in AeroDyn), fluid-inertia loads (from inflow acceleration in AeroDyn), and wave and current superposition (InflowWind + SeaState).

Best regards,
